src/consensus/aft/raft.h

```diff
@@ -889,32 +906,58 @@ namespace aft
           timeout_elapsed = 0ms;
 
           update_batch_size();
-          // Send newly available entries to all nodes.
-          for (const auto& it : nodes)
+          // Send newly available entries to all other nodes.
+          for (const auto& node : all_other_nodes)
           {
-            send_append_entries(it.first, it.second.sent_idx + 1);
+            send_append_entries(node.first, node.second.sent_idx + 1);
           }
         }
 
-        size_t backup_ack_timeout_count = 0;
-        for (auto& node : nodes)
+        for (auto& node : all_other_nodes)
         {
           node.second.last_ack_timeout += elapsed;
-          if (node.second.last_ack_timeout >= election_timeout)
+        }
+
+        bool has_quorum_of_backups = false;
+        for (auto const& conf : configurations)
+        {
+          size_t backup_ack_timeout_count = 0;
+          for (auto const& node : conf.nodes)
           {
-            backup_ack_timeout_count++;
+            auto search = all_other_nodes.find(node.first);
+            if (search == all_other_nodes.end())
+            {
+              // Ignore ourselves as primary
+              continue;
+            }
+            if (search->second.last_ack_timeout >= election_timeout)
+            {
+              LOG_DEBUG_FMT(
+                "No ack received from {} in last {}",
+                node.first,
+                election_timeout);
+              backup_ack_timeout_count++;
+            }
+          }
+
+          if (backup_ack_timeout_count < get_quorum(conf.nodes.size() - 1))
+          {
+            // If primary has quorum of active backups in any configuration, it
+            // should remain primary
+            has_quorum_of_backups = true;
+            break;
           }
         }
 
-        if (backup_ack_timeout_count >= get_quorum(nodes.size()))
+        if (!has_quorum_of_backups)
         {
           // CheckQuorum: The primary automatically steps down if it has not
-          // heard back from a majority of backups during an election timeout.
+          // heard back from a majority of backups in _all_ active
+          // configurations during an election timeout.
```

Comment:
This comment describes the opposite of the comment above, unless my close-reading skills are fried:

```
// If primary has quorum of active backups in any configuration, it
// should remain primary
```
```
// CheckQuorum: The primary automatically steps down if it has not
// heard back from a majority of backups in _all_ active
// configurations during an election timeout.
```

Are we 'supported' (ie - unwilling to step down as primary) by a quorum in a single configuration, or requiring a quorum in all configurations?

Reply:
I'd rather we switched to "each" everywhere possible, rather than "all", which is somewhat more vague.

Reply:
I've renamed this for clarity.